python/sglang/srt/model_executor/forward_batch_info.py

```diff
@@ -285,17 +318,32 @@ def update_flashinfer_indices(
 
         kv_indptr = torch.zeros((batch_size + 1,), dtype=torch.int32, device="cuda")
         kv_indptr[1:] = torch.cumsum(paged_kernel_lens, dim=0)
-        req_pool_indices_cpu = req_pool_indices.cpu().numpy()
-        paged_kernel_lens_cpu = paged_kernel_lens.cpu().numpy()
-        kv_indices = torch.cat(
-            [
-                model_runner.req_to_token_pool.req_to_token[
-                    req_pool_indices_cpu[i], : paged_kernel_lens_cpu[i]
-                ]
-                for i in range(batch_size)
-            ],
-            dim=0,
-        ).contiguous()
+
+        if True:
+            kv_indices = torch.empty(kv_indptr[-1], dtype=torch.int32, device="cuda")
+            create_kv_indices_triton[(batch_size,)](
+                model_runner.req_to_token_pool.req_to_token,
+                req_pool_indices,
+                paged_kernel_lens,
+                kv_indptr,
+                None,
+                model_runner.req_to_token_pool.req_to_token.size(1),
+                kv_indices,
+                BLOCK_SIZE=512,
+            )
+        else:
```

Comment:
It doesn't matter if this is not kept.

Reply:
The architectures we currently support include sm75, sm80, sm86, sm89, sm90. The supported CUDA versions include 11.8, 12.1, 12.4. The supported Torch versions include 2.2, 2.3, 2.4. Under this combination of configurations, we should ensure that Triton is working. In such a case, it seems unnecessary to retain the original logic of torch since here it is `if True` and will never be reached anyway.

Reply:
你的意思是不保留torch原生实现？

Reply:
你的意思是不保留torch原生实现？

Reply:
yes